reactor-core/src/test/java/reactor/core/publisher/UnicastProcessorTest.java

```diff
@@ -279,4 +279,50 @@ public void shouldSignalErrorOnOverflow() {
 		            .then(() -> processor.emitNext("boom"))
 		            .verifyErrorMatches(Exceptions::isOverflow);
 	}
+
+	@Test
+	public void tryEmitNextWithNoSubscriberAndBoundedQueueFailsZeroSubscriber() {
+		UnicastProcessor<Integer> unicastProcessor = UnicastProcessor.create(Queues.<Integer>one().get());
+
+		assertThat(unicastProcessor.tryEmitNext(1)).isEqualTo(Sinks.Emission.OK);
```

Comment:
IMO this is wrong and should return `FAIL_ZERO_SUBSCRIBER`

Reply:
the fact that `UnicastProcessor` allows to pre-emit (a few) elements before the actual subscription occurs is IMHO a cornerstone of it working so well as the plumbing for widowing operators. I fear doing what you suggest would open the door to all sorts of issues in windowing operators, at least.

Reply:
if so, the second emission should fail with OVERFLOW and not "zero subscribers" because the sink/processor isn't subscriber-dependent

Reply:
why? the whole point of adding this failed state is to indicate _undeliverable_ emission (as opposed to deliverable-in-case-of-later-subscription). there are still 0 subscribers when the second `tryEmitNext` occurs, but that one cannot be buffered for the benefit of a deferred subscriptions, so we return `FAIL_ZERO_SUBSCRIBER` (to distinguish from "there's a subscriber but it hasn't requested enough").

```java
		/**
		 * Has failed to emit the signal because the sink has never been subscribed to [AND]
         * has no capacity to buffer the signal.
		 */
		FAIL_ZERO_SUBSCRIBER;
```

(I should change the above wording to say `has currently no subscriber` maybe rather than `has never been subscribed)

Reply:
IMO the failure's cause is the overflow. The fact that it overflows when there are no subscribers is an implementation detail. Getting `FAIL_ZERO_SUBSCRIBER` after `OK` is super weird and IMO counterintuitive 😅